### PR TITLE
- Changed endpoint from form data to viewmodel and json

### PR DESCRIPTION
**Pull request by YanDav78**
- Fixed issue with dogFoundOn, was DateTime in DB instead of Date
- Changed resize_image_and_convert_to_format to not convert or resize if not needed
- Added missing predicates in build_filter
- Refactoring

### CODE REVIEW

File: app/DAL/repositories.py [hunk 1]
```diff
@@ -33,6 +34,12 @@ def add_dog_with_images(self, dogDTO: DogDTO) -> DogDTO:
                 return dogDTO
         except SQLAlchemyError as e:
             # Rollback transaction on error
+            logger.error(f"DB Error while adding dog with images: {e}")
```
Comment (SereneSound) [inline app/DAL/repositories.py]: @YanDav78 Can you change that to `logger.exception` so we have the stack trace automatically?

File: app/DAL/repositories.py [hunk 2]
```diff
@@ -33,6 +34,12 @@ def add_dog_with_images(self, dogDTO: DogDTO) -> DogDTO:
...
+            session.rollback()
+            raise e
+        except Exception as e:
+            # Rollback transaction on error
+            logger.error(f"Error while adding dog with images: {e}")
```
Comment (SereneSound) [inline app/DAL/repositories.py]: Here as well

**Review by SereneSound — APPROVED** [review]

File: app/DAL/repositories.py (hunk 1 above)
Comment (YanDav78) [reply app/DAL/repositories.py]: Done

File: app/DAL/repositories.py (hunk 2 above)
Comment (YanDav78) [reply app/DAL/repositories.py]: Done

**Pull request merged**